tests/integration/relations/test_backend_database.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+POSTGRESQL = "postgresql-k8s"
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.relations
+async def test_create_backend_db_admin_legacy_relation(ops_test: OpsTest):
+    """Test that the pgbouncer and postgres charms can relate to one another."""
+    # Build, deploy, and relate charms.
+    charm = await ops_test.build_charm(".")
+    resources = {
+        "pgbouncer-image": METADATA["resources"]["pgbouncer-image"]["upstream-source"],
+    }
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            charm,
+            resources=resources,
+            application_name=APP_NAME,
+        ),
+        # Edge 5 is the new postgres charm
+        ops_test.model.deploy(POSTGRESQL, channel="edge", trust=True, num_units=3),
+    )
+    await asyncio.gather(
+        ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000),
+        ops_test.model.wait_for_idle(
+            apps=[POSTGRESQL], status="active", timeout=1000, wait_for_exact_units=3
+        ),
+    )
+
+    await ops_test.model.relate(f"{APP_NAME}:backend-database", f"{POSTGRESQL}:database")
+    ops_test.model.wait_for_idle(apps=[APP_NAME, POSTGRESQL], status="active", timeout=1000),
```

Comment:
warning wait for idle might go through without waiting for relation hooks, leading to occassional failures and race conditions see [this convo](https://chat.canonical.com/canonical/pl/nh554o9gcfny8pcgu998bx759a)

Reply:
Cool suggestion, thanks!